lib-content-scrapers/src/jvmMain/kotlin/com/ustadmobile/lib/contentscrapers/folder/FolderIndexer.kt

```diff
@@ -0,0 +1,85 @@
+import org.kodein.di.DI
+import java.io.File
+import java.lang.Exception
+
+@ExperimentalStdlibApi
+class FolderIndexer(parentContentEntryUid: Long, runUid: Int, sqiUid: Int, contentEntryUid: Long, endpoint: Endpoint, di: DI) : Indexer(parentContentEntryUid, runUid, sqiUid, contentEntryUid, endpoint, di) {
+
+    override fun indexUrl(sourceUrl: String) {
+
+        try {
+
+            val folder = File(sourceUrl)
+
+            UMLogUtil.logInfo("folder generated ${folder.name}")
+
+            val fileList = folder.listFiles()
+
+            val name = folder.nameWithoutExtension
+
+            val folderEntry: ContentEntry
+            if (scrapeQueueItem?.overrideEntry == true && contentEntry != null) {
+
+                folderEntry = ContentScraperUtil.createOrUpdateContentEntry(contentEntry?.entryId.alternative(name), contentEntry?.title.alternative(name),
+                        folder.path, contentEntry?.publisher.alternative(""),
+                        contentEntry?.licenseType?.alternative(ContentEntry.LICENSE_TYPE_OTHER)
+                                ?: ContentEntry.LICENSE_TYPE_OTHER,
+                        contentEntry?.primaryLanguageUid?.alternative(englishLang.langUid)
+                                ?: englishLang.langUid,
+                        contentEntry?.languageVariantUid,
+                        ScraperConstants.EMPTY_STRING, false, contentEntry?.author ?: "", ScraperConstants.EMPTY_STRING, ScraperConstants.EMPTY_STRING,
+                        ScraperConstants.EMPTY_STRING, ContentEntry.TYPE_COLLECTION, contentEntryDao)
+
+            }else{
+
+                folderEntry = ContentScraperUtil.createOrUpdateContentEntry(name, name,
+                        folder.path, name, ContentEntry.LICENSE_TYPE_OTHER, englishLang.langUid, null,
+                        ScraperConstants.EMPTY_STRING, false, ScraperConstants.EMPTY_STRING, ScraperConstants.EMPTY_STRING, ScraperConstants.EMPTY_STRING,
+                        ScraperConstants.EMPTY_STRING, ContentEntry.TYPE_COLLECTION, contentEntryDao)
+
+            }
+
+
+
+            ContentScraperUtil.insertOrUpdateParentChildJoin(contentEntryParentChildJoinDao, parentcontentEntry, folderEntry, 0)
+
+            if (fileList == null || fileList.isEmpty()) {
+                close()
+                return
+            }
+
+            fileList.forEach { file ->
+
+                UMLogUtil.logInfo("found file ${file.name}")
+
+                val childEntry = ContentScraperUtil.insertTempContentEntry(contentEntryDao, file.path, folderEntry.primaryLanguageUid, file.name)
+                if (file.isDirectory) {
+                    createQueueItem(file.path, childEntry, ScraperTypes.FOLDER_INDEXER, ScrapeQueueItem.ITEM_TYPE_INDEX, folderEntry.contentEntryUid)
+                } else if (file.isFile) {
+                    createQueueItem(file.path, childEntry, ScraperTypes.FOLDER_SCRAPER, ScrapeQueueItem.ITEM_TYPE_SCRAPE, folderEntry.contentEntryUid)
```

Comment:
Why not check to see if the file extension is supported before creating a queue item?

Reply:
added